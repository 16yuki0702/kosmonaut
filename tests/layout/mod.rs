#[cfg(test)]
mod tests {
    use crate::util::{dump_layout_cmd, dump_layout_cmd_scaled};
    use insta::assert_snapshot;

    #[test]
    fn rainbow_divs_baseline() {
        let mut dump_layout_cmd = dump_layout_cmd();
        dump_layout_cmd
            .arg("--files")
            .arg("tests/websrc/rainbow-divs.html")
            .arg("tests/websrc/rainbow-divs.css")
            .succeeds()
            .no_stderr();
        assert_snapshot!(dump_layout_cmd.stdout());
    }

    #[test]
    fn rainbow_divs_baseline_two_scale_factor() {
        let mut dump_layout_cmd = dump_layout_cmd_scaled(2.0);
        dump_layout_cmd
            .arg("--files")
            .arg("tests/websrc/rainbow-divs.html")
            .arg("tests/websrc/rainbow-divs.css")
            .succeeds()
            .no_stderr();
        assert_snapshot!(dump_layout_cmd.stdout());
    }
}
