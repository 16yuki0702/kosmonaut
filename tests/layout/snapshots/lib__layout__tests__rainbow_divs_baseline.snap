---
source: tests/layout/mod.rs
expression: dump_layout_cmd.stdout()
---
DOCUMENT Inline LayoutBox at (0, 0) size 1920x184
  DOCTYPE Inline LayoutBox at (0, 0) size 1920x0
  HTML Block LayoutBox at (0, 0) size 1920x184
     AnonymousInline LayoutBox at (0, 0) size 1920x0
      TEXT Inline LayoutBox at (0, 0) size 1920x0
    BODY Block LayoutBox at (8, 8) size 1904x168
       AnonymousInline LayoutBox at (0, 0) size 1888x0
        TEXT Inline LayoutBox at (0, 0) size 1888x0
        TEXT Inline LayoutBox at (0, 0) size 1888x0
      DIV Block LayoutBox at (20, 20) size 1880x144
         AnonymousInline LayoutBox at (0, 0) size 1856x0
          TEXT Inline LayoutBox at (0, 0) size 1856x0
          TEXT Inline LayoutBox at (0, 0) size 1856x0
        DIV Block LayoutBox at (32, 32) size 1856x120
           AnonymousInline LayoutBox at (0, 0) size 1832x0
            TEXT Inline LayoutBox at (0, 0) size 1832x0
            TEXT Inline LayoutBox at (0, 0) size 1832x0
          DIV Block LayoutBox at (44, 44) size 1832x96
             AnonymousInline LayoutBox at (0, 0) size 1808x0
              TEXT Inline LayoutBox at (0, 0) size 1808x0
              TEXT Inline LayoutBox at (0, 0) size 1808x0
            DIV Block LayoutBox at (56, 56) size 1808x72
               AnonymousInline LayoutBox at (0, 0) size 1784x0
                TEXT Inline LayoutBox at (0, 0) size 1784x0
                TEXT Inline LayoutBox at (0, 0) size 1784x0
              DIV Block LayoutBox at (68, 68) size 1784x48
                 AnonymousInline LayoutBox at (0, 0) size 1760x0
                  TEXT Inline LayoutBox at (0, 0) size 1760x0
                  TEXT Inline LayoutBox at (0, 0) size 1760x0
                DIV Block LayoutBox at (80, 80) size 1760x24
                   AnonymousInline LayoutBox at (0, 0) size 1736x0
                    TEXT Inline LayoutBox at (0, 0) size 1736x0
                    TEXT Inline LayoutBox at (0, 0) size 1736x0
                  DIV Block LayoutBox at (92, 92) size 1736x0
                     AnonymousInline LayoutBox at (0, 0) size 1712x0
                      TEXT Inline LayoutBox at (0, 0) size 1712x0

